File: oppia/apps/user/services.py
# coding: utf-8
#
# Copyright 2013 Google Inc. All Rights Reserved.
#
# Licensed under the Apache License, Version 2.0 (the "License");
# you may not use this file except in compliance with the License.
# You may obtain a copy of the License at
#
#      http://www.apache.org/licenses/LICENSE-2.0
#
# Unless required by applicable law or agreed to in writing, software
# distributed under the License is distributed on an "AS-IS" BASIS,
# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
# See the License for the specific language governing permissions and
# limitations under the License.

"""Provides a seam for user-related services."""

__author__ = 'Sean Lip'


from google.appengine.api import users

def create_login_url(slug):
    """Creates a login url."""
    return users.create_login_url(slug)


def create_logout_url(slug):
    """Creates a logout url."""
    return users.create_logout_url(slug)


def is_current_user_admin():
    """Checks whether the current user is an admin."""
    return users.is_current_user_admin()


def get_current_user():
    """Returns the current user."""
    return users.get_current_user()

File: core/templates/dev/head/collection_editor/Services.js
// Copyright 2014 The Oppia Authors. All Rights Reserved.
//
// Licensed under the Apache License, Version 2.0 (the "License");
// you may not use this file except in compliance with the License.
// You may obtain a copy of the License at
//
//      http://www.apache.org/licenses/LICENSE-2.0
//
// Unless required by applicable law or agreed to in writing, software
// distributed under the License is distributed on an "AS-IS" BASIS,
// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
// See the License for the specific language governing permissions and
// limitations under the License.

/**
 * @fileoverview Standalone services for the exploration editor page.
 *
 * @author mgowano@google.com (Abraham Mgowano)
 */


oppia.factory('collectionData', ['$q', '$http', '$log', function($q, $http, $log) {
  // The pathname (without the hash) should be: .../create/{collection_id}
  var collectiononId = '';
  var pathnameArray = window.location.pathname.split('/');
  for (var i = 0; i < pathnameArray.length; i++) {
    if (pathnameArray[i] === 'create') {
      var collectionId = pathnameArray[i + 1];
      break;
    }
  }

  if(!collectionId) {
	  $log.error('Unexpected call to collection data for pathname ', collectionId);
    // Note: if we do not return anything, Karma unit tests fail.
    return {};
  }

  var collectionUrl = 'collection_editor/create/' + collectionId;
  var collectionDataUrl = '/collection_editor/data/' + collectionId;

  // Put collection variables here.
  var collectionData = {
    collectionId: collectionId,
    // Returns a promise that supplies the data for the current collection.
    getData: function() {
      if (collectionData.data) {
        $log.info('Found collection data in cache.');
      	var deferred = $q.defer();
	  	  deferred.resolve(collectionData.data);
      	return deferred.promise;
      } else {
      	// Retrieve data from the server.
      	return $http.get(collectionDataUrl).then(function(response) {
          $log.info('Retrieved collection data.');
          $log.info(response.data);
          collectionData.data = response.data;
          console.log(collectionData);
          return response.data;
        });
      }
    }
  }
  return collectionData;
}]);
